grails-app/jobs/org/pih/warehouse/jobs/AssignIdentifierJob.groovy

```diff
@@ -1,11 +1,24 @@
 package org.pih.warehouse.jobs
 
-import org.pih.warehouse.core.identification.BlankIdentifierResolver
-import org.pih.warehouse.product.Product
+import grails.gorm.transactions.Transactional
 
+import org.pih.warehouse.inventory.TransactionIdentifierService
+import org.pih.warehouse.order.OrderIdentifierService
+import org.pih.warehouse.product.ProductIdentifierService
+import org.pih.warehouse.receiving.ReceiptIdentifierService
+import org.pih.warehouse.requisition.RequisitionIdentifierService
+import org.pih.warehouse.shipping.ShipmentIdentifierService
+
+@Transactional
```

Comment:
I had to add this to make the job work. Without it it was hitting "no transaction is in progress" errors. Maybe a better solution though is to wrap each individual generateForAllUnassignedIdentifiers() call in their own transaction based on the underlying table that they're working on.

@awalkowiak I see you had this PR relating to transactions in jobs so maybe you'll know better than me https://github.com/openboxes/openboxes/pull/4160

Reply:
Possibly relevant since we are doing `sessionRequired=false` in all our jobs: https://grails.github.io/grails-quartz/guide/configuration.html

> Hibernate Sessions and Jobs
> Jobs are configured by default to have Hibernate Session bounded to thread each time job is executed. This is required if you are using Hibernate code which requires open session (such as lazy loading of collections) or working with domain objects with unique persistent constraint (it uses Hibernate Session behind the scene). If you want to override this behavior (rarely useful) you can use 'sessionRequired' property:
> 
> def sessionRequired = false

Reply:
@EWaterman here you might find a bit more details about the issue with `no transaction is in progress` https://pihemr.atlassian.net/browse/OBGM-584 (this is the OG ticket under which I introduced the wrapping service calls with the `withNewSession`), this was basically the same issue you run into here: https://github.com/openboxes/openboxes/pull/4967#issuecomment-2512574451, back then I remember `@Transactional` did not work for me. Does the `@Transactional` work now? Are the new identifiers persisted to the db correctly now?